### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by ashwath462**

### CODE REVIEW

File: models/ItemFactory.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { RawItem } from "./RawItem";
+import { ManufacturedItem } from "./ManufacturedItem";
+import { ImportedItem } from "./ImportedItem";
+
+export class ItemFactory {
+    public static createItem(userInput: any) {
+        try {
+            let item: any;
+            switch (userInput.get('type')) {
+                case 'raw':
```
Comment (piyushmalviya13) [inline models/ItemFactory.ts:11]: Use enum here

File: models/Item.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export abstract class Item {
+    protected name: string;
+    protected price: number;   //original price
+    protected quantity: number;
+    protected salesTax: number;
+    protected finalPrice: number;  //price after applying tax, taxedPrice
+    protected type: string;
```
Comment (piyushmalviya13) [inline models/Item.ts:7]: Make use of the type enum created

File: utility/utils.ts [hunk 3]
```diff
@@ -1,3 +1,97 @@
...
+
+    if (itemValues.length == 8) {
+        itemValues.map((e: string) => { e = e.toLowerCase() });
+        return mapInputDetails(itemValues);
+    }
+
+    throw new Error("Kindly enter valid details");
+}
+
+export function saveItem(item: Item, allItemsList: Item[]): Item[] {
```
Comment (piyushmalviya13) [inline utility/utils.ts:74]: Cover these functions as well in unit tests

**Review by piyushmalviya13 — CHANGES_REQUESTED** [review]
Look into the comments

File: index.ts [hunk 4]
```diff
@@ -2,6 +2,6 @@ import { question } from 'readline-sync';
 import { add } from './utility/utils';
```
[the remark this answers is not in the document]
Comment (ashwath462) [reply index.ts:2]: Done

File: models/ItemFactory.ts (hunk 1 above)
Comment (ashwath462) [reply models/ItemFactory.ts:11]: Done

File: models/Item.ts (hunk 2 above)
Comment (ashwath462) [reply models/Item.ts:7]: Done

**Review by piyushmalviya13 — APPROVED** [review]
Looks good

**Pull request merged**